ui/app/routes/vault/cluster/secrets/backend/secret-edit.js

```diff
@@ -107,7 +107,7 @@ export default Route.extend(UnloadModelRoute, {
       pki: secret && secret.startsWith('cert/') ? 'pki/cert' : 'pki/pki-role',
       cubbyhole: 'secret',
       kv: backendModel.get('modelTypeForKV'),
-      keymgmt: `keymgmt/${options.queryParams?.itemType || 'key'}`,
+      keymgmt: options.queryParams?.itemType === 'provider' ? 'keymgmt/provider' : 'keymgmt/key',
```

Comment:
Curious if `itemType` can be anything other than _provider_, _key_ or null/undefined? Was the view broken in some way from the original line?

Reply:
Sorry jumped straight to file changes and missed the screen shots! Thanks for the update but just curious what the model was resolving to?

Reply:
Yes, `itemType` can be the `key` type. For example, `rsa-2048`. It was a causing a `newModel.proto(...).getHelpUrl is not a function` error since in the `pathHelp` service the model was `model:keymgmt/rsa-2048` from the`modelType`. I couldn't reproduce an area where `itemType` was `null/undefined`. Any thoughts on what _may_ cause `itemType` to be undefined?

Reply:
Hmm that seems like a bug somewhere else then since the `itemType` query param represents which tab to display so it should only be set to either `provider` or `key`. It looks like the issue is with the `secret-query-params` helper. In the `secret-list/item` item component it is setting it to `type` which must be the key type.

Reply:
Ahh that makes sense thanks for catching that!